test/node/bson_undefined.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expect } from 'chai';
+import { bufferFromHexArray } from './tools/utils';
+import { BSON } from '../register-bson';
+import { BSON_DATA_NULL } from '../../src/constants';
+
+describe('BSON undefined', () => {
+  const KEY_A = '6100';
+  const KEY_0 = '3000';
+  const KEY_1 = '3100';
+  const KEY_2 = '3200';
+  const bsonDocWithUndefined = bufferFromHexArray([
+    '06', // BSON undefined
+    KEY_A
+  ]);
+
+  describe('when deserialize is given BSON bytes with undefined value', function () {
+    it('returns a javascript undefined value', () => {
+      const doc = BSON.deserialize(bsonDocWithUndefined);
+      expect(doc).to.have.own.property('a').that.is.undefined;
+    });
+  });
+
+  describe('when serialize is given a javascript object that contains undefined', () => {
+    describe('when ignoreUndefined is set to false', function () {
+      it('serializes to document with a set to BSON null (type=10)', () => {
+        const jsObject = BSON.deserialize(bsonDocWithUndefined);
+        const bytes = BSON.serialize(jsObject, { ignoreUndefined: false });
+        expect(bytes).to.have.lengthOf(8);
+        expect(bytes).to.have.own.property('4', BSON_DATA_NULL);
```

Comment:
What is happening in this assertion?  Does `haveOwnProperty` check for index access if the element is a Buffer?

Could we use parseToElementsToArray to re-parse the serialized bytes and confirm the serialized bytes contain one BSON null?

Reply:
good use of parseToElements